lib/chef/knife/bootstrap.rb

```diff
@@ -211,6 +225,47 @@ def read_template
         IO.read(@template_file).chomp
       end
 
+      def normalized_run_list
+        case config[:run_list]
+        when nil
+          []
+        when String
+          config[:run_list].split(/\s*,\s*/)
+        when Array
+          config[:run_list]
+        end
+      end
+
+      def generate_client_pem(node_name)
+        tmpdir = Dir.mktmpdir
```

Comment:
We'll need to clean this up someplace below or use the form of mktmpdir where you give it a block.